### PR TITLE
feat(cart): add non empty cart guard

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

**Comment by damienwebdev** [comment]
The naming here is a bit odd, what does a NonEmptyCartGuard do? Does it guard against any empty cart, does it guard to ensure that the cart is empty?

**Comment by griest024** [comment]
@damienwebdev  It guards against an empty cart, i.e., it will emit `true` if the cart has items in it, `false` if its empty. I agree the name could be better; I'm open to suggestions.

**Comment by lderrickable** [comment]
What about just `EmptyCartGuard`? When I think of "guard", I think of something guarding one thing against something else. The "one thing" is just the page that is being routed, so I think the naming of any guard should be the "something else" being guarded against, which would be an empty cart. So `EmptyCartGuard`

**Comment by griest024** [comment]
@lderrickable hm well I don't disagree but our other guards really aren't named like that.

**Comment by lderrickable** [comment]
lol true

**Comment by lderrickable** [comment]
I guess we could either call this one `DaffCartItemsGuard` or change the other guards to `DaffEmptyBillingAddressGuard`

**Comment by griest024** [comment]
I vote for `DaffCartItemsGuard`

**Comment by damienwebdev** [comment]
DaffVerifyCartHasItemsGuard?

**Comment by griest024** [comment]
> DaffVerifyCartHasItemsGuard?

While I'm not totally against that, I prefer `DaffCartItemsGuard`. Its the same pattern as our other guards. Whatever we choose should be consistent across all of the guards.

**Comment by damienwebdev** [comment]
`DaffCartItemsGuard` is fine along with a nice doc-block.

**Comment by griest024** [comment]
changed

File: libs/cart/state/src/guards/non-empty-cart/non-empty-cart.guard.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import { CanActivate, Router } from '@angular/router';
```
Comment (lderrickable) [inline libs/cart/state/src/guards/non-empty-cart/non-empty-cart.guard.ts:1]: The file name and path needs to be changed to `cart-item` from `non-empty-cart`

Comment (griest024) [reply libs/cart/state/src/guards/non-empty-cart/non-empty-cart.guard.ts:1]: good catch, changed

**Review by lderrickable — APPROVED** [review]

**Review by damienwebdev — APPROVED** [review]
Lgtm

**Pull request merged**